### PR TITLE
Add core.ac.uk engine

### PR DESCRIPTION
**Pull request by marc-cenon**
## What does this PR do?

Add a search engine for core.ac.uk  : https://core.ac.uk/

## Why is this change important?

Core is the world’s largest collection of open access research papers
Plus it has a lot of options for text mining

## How to test this PR locally?

First get your API key following this page: https://core.ac.uk/api-keys/register/
In searx/engines/core.py :
I left my own key, commented out on line 28. Feel free to use it, if needed.

In searx in the science category, try search using the core.ac.uk engine.

In this engine, not all the published dates are correct. When there is no date linked to an article in the engine, the search result will display " 0 minute ago" for the published date

### CODE REVIEW

**Review by kvch — APPROVED** [review]

**Comment by kvch** [comment]
@spongebob33 Could you please rebase your PR?

**Comment by return42** [comment]
@spongebob33 LGTM thanks a lot. If you interested in, in my branch https://github.com/return42/searx/commits/core.ac.uk you will find commit https://github.com/return42/searx/commit/16c2615204df81f89feae943fa668ca23a3fcbd9

If you like, cherry pick from .. or otherwise I can create a new PR after this PR has been merged.

**Comment by return42** [comment]
@kvch do you like to merge this PR ? / I suppose @spongebob33 is absent from github?

> @spongebob33 Could you please rebase your PR?

I'm not a github expert but AFAIK maintainers are able to [rebase and merge PR](https://github.blog/2016-09-26-rebase-and-merge-pull-requests/)

File: searx/engines/core.py
Comment (dalf) [inline searx/engines/core.py:14]: The wikidata Id is Q22661180

**Comment by asciimoo** [comment]
@spongebob33 could you address the review note from @dalf and comment from @kvch? Also, it would be probably good to disable it by default since it requires an API key which isn't the best from a privacy viewpoint. What do you think?

**Comment by kvch** [comment]
Added in https://github.com/searx/searx/commit/6513a5606431b5ab4fbe6eefb70510ba47ab5e04
Thank you!

**Pull request closed without merging**
